### PR TITLE
Release/0.11.1

### PR DESCRIPTION
**Pull request by igable**
This PR has everything needed for multiple endpoint. Feedback would be particularly useful on the commits from the develop branch up to the release: [develop...release/0.11.1](https://github.com/enStratus/mixcoatl/compare/develop...release/0.11.1)

The details of multiple endpoint support pull request are [here](https://github.com/enStratus/mixcoatl/pull/225), and you can see details of how to test it [here](https://gist.github.com/igable/70b49702613bb7462da3). The [release notes](https://github.com/enStratus/mixcoatl/blob/4d50858e0fef19f6fee78c853c8ce74d5d50a32f/RELEASENOTES.md) contain examples of how you might use multiple endpoint support.

### CODE REVIEW

File: mixcoatl/resource.py [hunk 1]
```diff
@@ -8,6 +8,105 @@
...
+        try:
+            datetime.datetime.strptime(date, '%Y-%m-%d')
+            return date
+        except ValueError:
+            raise ValueError("Incorrectly formatted API version, should be YYYY-MM-DD")
+
+    @classmethod
+    def from_file(cls, filename):
+        """ Load a DCM endpoint definition from a json file and return and Endpoint object. For an example file see:
+            TODO: add github pointed to example file
```
Comment (oldpatricka) [inline mixcoatl/resource.py]: Did you mean to leave this as a TODO?

File: mixcoatl/resource.py [hunk 2]
```diff
@@ -8,6 +8,105 @@
...
+                        secret_key=str(e['secret_key']),
+                        access_key=str(e['access_key']),
+                        ssl_verify=e['ssl_verify'] if 'ssl_verify' in e else None,
+                        nickname=e['nickname'] if 'nickname' in e else None)
+
+    @classmethod
+    def multiple_from_file(cls, filename):
+        """ Load multiple DCM endpoint definitions from a json file and return a dictionary indexed by endpont nicknames
+        For an example file see:
+            TODO: add github pointed to example file
```
Comment (oldpatricka) [inline mixcoatl/resource.py]: Same here

File: tests/live/test_server.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    def test_saas_server_create(self):
+        """Test if we can create a VM using an live endpoint object and shut it down"""
+
+        # This cli command would make a server similar to below
+        # dcm-create-server --machineimage 29710 --datacenter 1273 \
+        #     --name igable-terminateme --productid t1.micro --budgetid 1512 --description longstring
+
+        self.server = Server(endpoint=self.endpoint)
+        self.server.set_from_file("../../../secret-test-data/ubuntu-trusty-14.04-amd64-server-20150629.json")
```
Comment (oldpatricka) [inline tests/live/test_server.py]: Would be neat if this printed a helpful error explaining where to set this file and maybe skipped the test if not present.

**Comment by oldpatricka** [comment]
A couple optional suggestions, otherwise :+1:

File: mixcoatl/resource.py (hunk 2 above)
Comment (igable) [inline mixcoatl/resource.py]: fixed in b9fbeff

File: mixcoatl/resource.py (hunk 1 above)
Comment (igable) [inline mixcoatl/resource.py]: fixed in b9fbeff

File: tests/live/test_server.py (hunk 3 above)
Comment (igable) [inline tests/live/test_server.py]: Sounds like a good idea. Were you somehow able to trigger this test without specifically calling it?

**Comment by igable** [comment]
Closing in favor of  #240, however the comments made by @oldpatricka will be fixed in #240.

File: tests/live/test_server.py (hunk 3 above)
Comment (igable) [inline tests/live/test_server.py]: Discussion with Patrick that this test wasn't automatically triggered, hence we are going to wait for the next release to fix this. Issue is #242, expect for release 1.1.0.

**Pull request closed without merging**